test/testcomplexity.cpp

```diff
@@ -0,0 +1,128 @@
+
+#include "abc.h"
+#include "constant.h"
+#include "power.h"
+#include "sum.h"
+#include "product.h"
+#include "trigonometric.h"
+#include "tsymtests.h"
+#include "undefined.h"
+#include "logarithm.h"
+
+using namespace tsym;
+
+TEST_GROUP(Complexity)
+{
+    
+    BasePtr undef;
+    BasePtr integer;
+    BasePtr fraction;
+    BasePtr doubleval;
+    BasePtr constant;
+    BasePtr symbol;
+    BasePtr sum;
+    BasePtr prod;
+    BasePtr power;
+    BasePtr sinus;
+    BasePtr ln;
+    BasePtr undefnumeric;
+    void setup()
+    {
+        double a =2.7665454894445454;
+        undef = Undefined::create();
+        integer = Numeric::create(3);
+        fraction = Numeric::create(2,3);
+        doubleval= Numeric::create(a);
+        constant = Constant::createPi();
+        symbol = Symbol::create("a");
+        sum = Sum::create(integer,symbol);
+        prod = Product::create(integer,symbol,constant);
+        power = Power::create(integer,symbol);
+        sinus = Trigonometric::createSin(symbol);
+        ln = Logarithm::create(symbol);
+    }
+
+};
+
+TEST(Complexity, Undefined)
+{
+    unsigned res = undef->complexity();
+
+    CHECK_EQUAL(0,res);
+}
+
+TEST(Complexity, Integer)
+{
+    unsigned res = integer->complexity();
+
+    CHECK_EQUAL(1,res);
+}
+
+TEST(Complexity, Fraction)
+{
+    unsigned res = fraction->complexity();
+
+    CHECK_EQUAL(2,res);
+}
+
+TEST(Complexity, Double)
+{
+    unsigned res = doubleval->complexity();
+
+    CHECK_EQUAL(3,res);
+}
+
+TEST(Complexity, Constant)
+{
+    unsigned res = constant->complexity();
+
+    CHECK_EQUAL(4,res);
+}
+
+TEST(Complexity, Symbol)
+{
+    unsigned res = symbol->complexity();
+
+    CHECK_EQUAL(5, res);
+}
+
+TEST(Complexity, Sum)
+{   
+    unsigned expected = 5+1+5;
```

Comment:
i guess it's ok to simply go on with CHECK_EQUAL(11, sum->complexity), you don't have to use local variables for everything. If it's readable enough and the local variable names don't add the necessary context to understand the test, then keep it short. applies to the next test, too, of course.